lib/kdtree.h

```diff
@@ -219,6 +221,36 @@ class FlatNode {
         return static_cast<uint32_t>(data_) >> 2;
     }
 
+    // Since right nodes are create after parent nodes, we need a setter to
+    // update the index of the right node after creation.
+    void set_right(uint32_t value) {
+        data_ = (data_ & 0x0000) |
```

Comment:
This was a bug :) It should be `data_ & 0xFFFFFFFF00000000`, to clean-up all lower bits.